src/lib/SX1280Driver/SX1280.cpp

```diff
@@ -67,27 +70,47 @@ bool SX1280Driver::Begin()
     hal.WriteCommand(SX1280_RADIO_SET_AUTOFS, 0x01);                                                                              //Enable auto FS
     hal.WriteRegister(0x0891, (hal.ReadRegister(0x0891) | 0xC0));                                                                 //default is low power mode, switch to high sensitivity instead
-    SetPacketParams(12, SX1280_LORA_PACKET_IMPLICIT, 8, SX1280_LORA_CRC_OFF, SX1280_LORA_IQ_NORMAL);                              //default params
+    SetPacketParamsLoRa(12, SX1280_LORA_PACKET_IMPLICIT, 8, SX1280_LORA_CRC_OFF, SX1280_LORA_IQ_NORMAL);                          //default params
     SetFrequencyReg(currFreq);                                                                                                    //Set Freq
     SetFIFOaddr(0x00, 0x00);                                                                                                      //Config FIFO addr
-    SetDioIrqParams(SX1280_IRQ_RADIO_ALL, SX1280_IRQ_TX_DONE | SX1280_IRQ_RX_DONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE); //set IRQ to both RXdone/TXdone on DIO1
+    SetDioIrqParams(SX1280_IRQ_RADIO_ALL, SX1280_IRQ_TX_DONE | SX1280_IRQ_RX_DONE);                                               //set IRQ to both RXdone/TXdone on DIO1
 #if defined(USE_SX1280_DCDC)
     hal.WriteCommand(SX1280_RADIO_SET_REGULATORMODE, SX1280_USE_DCDC);     // Enable DCDC converter instead of LDO
 #endif
     return true;
 }
 
-void SX1280Driver::Config(SX1280_RadioLoRaBandwidths_t bw, SX1280_RadioLoRaSpreadingFactors_t sf, SX1280_RadioLoRaCodingRates_t cr, uint32_t freq, uint8_t PreambleLength, bool InvertIQ, uint8_t PayloadLength)
+void SX1280Driver::Config(uint8_t bw, uint8_t sf, uint8_t cr, uint32_t freq,
+                          uint8_t PreambleLength, bool InvertIQ, uint8_t _PayloadLength,
+                          uint32_t flrcSyncWord, uint16_t flrcCrcSeed, uint8_t flrc)
 {
-    PayloadLength = PayloadLength;
+    uint8_t irqs = SX1280_IRQ_TX_DONE | SX1280_IRQ_RX_DONE;
+    uint8_t const mode = (flrc) ? SX1280_PACKET_TYPE_FLRC : SX1280_PACKET_TYPE_LORA;
+
+    PayloadLength = _PayloadLength;
     IQinverted = InvertIQ;
     SetMode(SX1280_MODE_STDBY_XOSC);
-    ConfigLoRaModParams(bw, sf, cr);
-    SetPacketParams(PreambleLength, SX1280_LORA_PACKET_IMPLICIT, PayloadLength, SX1280_LORA_CRC_OFF, (SX1280_RadioLoRaIQModes_t)((uint8_t)!IQinverted << 6)); // TODO don't make static etc. LORA_IQ_STD = 0x40, LORA_IQ_INVERTED = 0x00
+    hal.WriteCommand(SX1280_RADIO_SET_PACKETTYPE, mode);
+    if (mode == SX1280_PACKET_TYPE_FLRC)
+    {
+        DBGLN("Config FLRC");
+        ConfigModParamsFLRC(bw, cr, sf);
+        SetPacketParamsFLRC(SX1280_FLRC_PACKET_FIXED_LENGTH, /*crc=*/1,
+                            PreambleLength, _PayloadLength, flrcSyncWord, flrcCrcSeed);
+        irqs |= SX1280_IRQ_CRC_ERROR;
```

Comment:
@CapnBry Do you have idea how to handle this CRC error? 
Should I just ignore ISR (as it is handled currently) or `RXdoneCallback(crcError = true);` ?

Reply:
If we detect a CRC error in a Lora packet, we just drop it and move on without adjusting the sync or anything. There's a chance of getting someone else's packet and it being detected as valid but with a bad CRC which could throw off our timing, so I'm not sure we should do anything if the hardware detects a CRC error.

That said, for testing purposes, it would be good to have it throw an event and just set the `lastPacketCrcError` flag in rx_main to make the scoreboard look correct?

Reply:
Ok, so it would make sense to handle that IRQ... I will take a look how to handle it properly